llvm/include/llvm/ProfileData/InstrProf.h

```diff
@@ -565,6 +674,13 @@ void InstrProfSymtab::finalizeSymtab() {
   llvm::sort(AddrToMD5Map, less_first());
   AddrToMD5Map.erase(std::unique(AddrToMD5Map.begin(), AddrToMD5Map.end()),
                      AddrToMD5Map.end());
+  // GlobalVariable address ranges should not overlap; so sort by either
```

Comment:
I think this is true for your vtable use case but not in general? E.g. Readonly string and other primitive constants can be merged such that the symbol names point to the same address range.

Reply:
> Readonly string and other primitive constants can be merged such that the symbol names point to the same address range.

This makes sense. I think a similar example is a global alias and its corresponding object. Update the comment to 'VTable objects address ranges shouldn't overlap'